### ISSUE TITLE
Percent-sign followed by only one character leads to `StringIndexOutOfBoundsException` when appearing at end of component (e.g., path or query-var)

### ISSUE THREAD

**Issue by JustAHappyKid** [lemonlabsuk/scala-uri#1]
It should be raising a `UriDecodeException` (presumably), as it would do with an invalid `%` sequence, like `%zz` for example.

Here's a test-case:

```
  it should "give proper exception when percent-sign is followed by less than 2 characters" in {
    Seq("/?x=%3", "/%3", "/?a=%3&b=whatever", "/?%3=okay").foreach { part =>
      try Uri.parse("http://test.com" + part)
      catch { case _: com.netaporter.uri.decoding.UriDecodeException => /* No problem! */}
    }
  }
```

**Comment by theon** [comment]
Thanks for raising this bug 👍 

I've pushed a fix to master and will cut a release shortly.

**Comment by JustAHappyKid** [comment]
Sounds good -- thanks. :-)

**Comment by mkuthan** [comment]
I observed "StringIndexOutOfBoundsException: String index out of range: 2" exception in my application and found this issue. Could you cut a new release, please?

**Comment by theon** [comment]
@mkuthan Released `0.4.17`

**Comment by mkuthan** [comment]
Thank you!
